### PR TITLE
feat: add data viz section to website

### PR DESCRIPTION
**Pull request by alisonjoseph**
# Add Data Visualization section to site

closes: #240, #241, #364, #242, #243, #387, #351

## Dev review instructions 

A ton of files added because we are adding a whole new section to the left nav. A lot of them are images so do not need review. Please verify all the components used in the data viz pages are working:
- Tag component on Overview cards (new)
- InlineNotification component (new)
- Live chart demos on `Basic charts` and `Advanced charts` page
- Verify that all links and anchor links are working correctly
- Verify that all pages can be searched in search bar
- Verify that image / ArtDirection component display and response to breakpoints

We don't expect dev review to go into the content, but obviously if you see a problem with content, definitely tell us. Any additional feedback welcomed!

## Design review instructions

TBD - working with Mike in person on this at the moment. 

## Content review instructions

FYI @connor-leech @janchild 

- Make sure there are no conflicting guidance
- Clear messaging (eg. inline notification message)
- Heading, content, images and captions match
- Language, word choice, spelling, grammar help, including texts inside images

### CODE REVIEW

**Comment by shixiedesign** [comment]
This branch closes: https://github.com/carbon-design-system/carbon-website/issues/240, https://github.com/carbon-design-system/carbon-website/issues/241

**Comment by shixiedesign** [comment]
@alisonjoseph Could you help with a few changes to the nav? Since we are anticipating the tabs to go away, we'd like to break out `Overview`'s 2 tabs and `Charts`'s 2 tabs all into individual pages. This is the nav for Oct: 

**Data visualization**
- Getting started (formally Develop page, just a name change. Keeping tabs here.) 
- Chart types
- Anatomy (formally Charting elements page)
- Basic charts
- Advanced charts
- Color palettes
- Axis and labels
- Legends
- Tooltips

THANKS!

**Comment by connor-leech** [comment]
@shixiedesign I can update the nav!

**Comment by alisonjoseph** [comment]
Thanks Connor! let me know if I can help

**Comment by shixiedesign** [comment]
Hi @alisonjoseph, got a few items for to get your eyes on before you head out: 

1. Most important, is could you work with Eliad to pull in the latest chart demos? As shown here: https://carbon-design-system.github.io/carbon-charts/ If refactor can be done later Eliad can try to update it even more, but at this point, even updating to the versions in the link above would help! Please make sure all charts are on a white background too. It's hard to get over the visuals differences for a lot of people. 

2. Spacing around inline notification is too big. Ideally it can be closer to the element above it: 
image

3. Cards in Chart types no longer link to the charts page's corresponding H2s. This is probably because I asked Connor to change the nav. I think I can fix this if you are out of time. Just FYI. 

Thank you!

**Comment by alisonjoseph** [comment]
made an issue to fix the spacing https://github.com/carbon-design-system/gatsby-theme-carbon/issues/453

**Comment by alisonjoseph** [comment]
@shixiedesign latest React demos are here https://carbon-design-system.github.io/carbon-charts/react/?path=/story/bar--label-based-legend thats what is being pulled into the website. I don't think I'll have time to try and update them today to match the link you have, not sure if @theiliad will?

**Comment by shixiedesign** [comment]
Okay thanks! lemme check with Eliad. I think that's the last important thing we hopefully can do before shipping this whole section. I don't think we should publish this section until the charts demos are the new style. If it has to wait till next week, we will do that.

**Comment by theiliad** [comment]
@shixiedesign Do we want to have the sketch link show up before the other 2 dev-related links?
image

Since this is a developer get started page I think the sketch should be number 3 in the list

**Comment by theiliad** [comment]
Also, we should add it to the other framework pages as well.

Also, in the "other frameworks" page, the title should be `Get Started` not `Getting Started`

File: src/pages/data-visualization/legends/index.mdx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+![legends behavior 4](images/legends-behavior-4.png)
+![legends behavior 4](images/legends-behavior-4.png)
+
+</ArtDirection>
+</Column>
+</Row>
+
+### Hidden legends
+
+Please note that hiding legends is discouraged in data visualization (unless only 1 category of data is displayed). This design is for mobile situations where having legends at a glance is less essential. In general, hiding legends reduces the clarity of the visualization and is inaccessibility.
```
Comment (jendowns) [inline src/pages/data-visualization/legends/index.mdx]:
```suggestion
Please note that hiding legends is discouraged in data visualization (unless only one category of data is displayed). This design is for mobile situations where having legends at a glance is less essential. In general, hiding legends reduces the clarity of the visualization and is inaccessible.
```

File: src/pages/data-visualization/advanced-charts/index.mdx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+</Row>
+
+## Heatmap
+
+A heat map is a graphical representation of data where the individual values contained in a matrix are represented as colors.
+
+<Row>
+<Column  colLg={8} colMd={6} colSm={4} >
+
+![Horizontal bar chart](images/charts-adv-heatmap.png)
```
[the remark this answers is not in the document]
Comment (shixiedesign) [reply src/pages/data-visualization/advanced-charts/index.mdx]: Yep.

**Comment by alisonjoseph** [comment]
This all looks good to me! (I can't officially approve since its technically my PR). 

One comment, did we want to add a link/card to Carbon-charts on the Resources page?

File: src/pages/data-visualization/getting-started/react.mdx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+## Development
+
+Please refer to the [Contribution Guidelines](https://github.com/carbon-design-system/carbon-charts/blob/master/CONTRIBUTING.md) before starting any work.
+
+### Using the server
+
+We recommend the use of [React Storybook](https://github.com/storybookjs/storybook/tree/next/app/react) for developing components.
+
+#### Start the server:
```
Comment (jnm2377) [inline src/pages/data-visualization/getting-started/react.mdx]: Is it just `h4`'s that are getting a semicolon? Not sure if this was intentional or not. The other headers don't have one.

File: src/pages/data-visualization/getting-started/vanilla.mdx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+</Column>
+</Row>
+
+## Carbon Charts Vanilla library
+
+The library provides front-end developers & engineers a collection of reusable charting components to build websites and user interfaces. Adopting the library enables developers to use consistent markup, styles, and behavior in prototype and production work.
+
+## Install
+
+#### Using yarn:
```
Comment (jnm2377) [inline src/pages/data-visualization/getting-started/vanilla.mdx]: same as before about `:`

File: src/pages/data-visualization/getting-started/vanilla.mdx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+<br />
+
+```bash
+$ yarn add @carbon/charts d3
+```
+
+#### If you prefer npm:
```
Comment (jnm2377) [inline src/pages/data-visualization/getting-started/vanilla.mdx]: `:`?

File: src/pages/data-visualization/getting-started/vue.mdx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  </Column>
+</Row>
+
+## Carbon Charts Vue library
+
+The library provides front-end developers & engineers a collection of reusable charting components to build websites and user interfaces. Adopting the library enables developers to use consistent markup, styles, and behavior in prototype and production work.
+
+## Install
+
+#### Using yarn:
```
Comment (jnm2377) [inline src/pages/data-visualization/getting-started/vue.mdx]: same question as before, feel free to ignore

File: src/pages/data-visualization/getting-started/vue.mdx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+<br />
+
+```bash
+$ yarn add @carbon/charts @carbon/charts-vue d3
+```
+
+#### If you prefer npm:
```
Comment (jnm2377) [inline src/pages/data-visualization/getting-started/vue.mdx]: same

File: src/pages/data-visualization/getting-started/vue.mdx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+## Development
+
+Please refer to the [Contribution Guidelines](https://github.com/carbon-design-system/carbon-charts/blob/master/CONTRIBUTING.md) before starting any work.
+
+### Using the server
+
+We recommend the use of [Vue Storybook](https://github.com/storybookjs/storybook/tree/next/app/vue) for developing components.
+
+#### Start the server:
```
Comment (jnm2377) [inline src/pages/data-visualization/getting-started/vue.mdx]: same

File: src/pages/data-visualization/data-visualization.scss [hunk 9]
```diff
@@ -0,0 +1,4 @@
+div.chart-holder {
+  background: #fff;
+  padding: 30px;
```
[the remark this answers is not in the document]
Comment (alisonjoseph) [reply src/pages/data-visualization/data-visualization.scss]: 👍 yes should also use color token

File: src/pages/data-visualization/color-palettes/index.mdx [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+![Diverging palette 2](images/palettes-div2.png)
+![Diverging palette 2](images/palettes-div2.png)
+
+</ArtDirection>
+</Column>
+</Row>
+
+## Alert palette
+
+Alert colors are used to indicate status. Typically, red represent danger or error; orange represent serious warning; yellow represent regular warning, and green represent normal or success.
```
[the remark this answers is not in the document]
Comment (alisonjoseph) [reply src/pages/data-visualization/color-palettes/index.mdx]: cc @connor-leech

File: src/pages/data-visualization/getting-started/vanilla.mdx (hunk 4 above)
Comment (alisonjoseph) [reply src/pages/data-visualization/getting-started/vanilla.mdx]: @connor-leech ?

File: src/pages/data-visualization/color-palettes/index.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (connor-leech) [reply src/pages/data-visualization/color-palettes/index.mdx]: All of these will be fixed in a sec, I just did a spelling pass

File: src/pages/data-visualization/getting-started/vanilla.mdx (hunk 4 above)
Comment (connor-leech) [reply src/pages/data-visualization/getting-started/vanilla.mdx]: We shouldn't have colons in headers, good call @jnm2377

**Pull request merged**